Appendix/process (#42)
MrLongbottom/P9-Rapport: sections/appendix/process.tex

\subsection{Project Process}
\begin{itemize}
	\item Initial Idea - Combine LDA and PageRank (clustering)
	\item Preprocessing \& LDA implementation
	\item Adjacency matrix construction
	\begin{itemize}
		\item challenges with calculation time
		\item focus on more restricting preprocessing and thresholds
	\end{itemize}
	\item Bad results
	\item Discovery of related papers
	\item Found out our model didn't converge
\end{itemize}

> Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the information retrival results of \gls{lda}.

Reviewer: ```suggestion
Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the information retrieval results of \gls{lda}.
```

We began by implementing some initial preprocessing methods and using an \gls{lda} implementation by gensim\todo{ref}.

> After this we only needed to combine \gls{lda} with a PageRank implementation. But in order to do so, we first needed to construct an adjacency matrix. This is where we started to run into trouble as constructing a large adjecency matrix based a similarity measure took extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.

Reviewer: ```suggestion
After this, we only needed to combine \gls{lda} with a PageRank implementation. 
But to do so, we first needed to construct an adjacency matrix. 
This is where we started to run into trouble as constructing a large adjacency matrix based on a similarity measure took an extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.
```

> We made the preprocessing more aggressive and implemeneted more preprocessing methods, to contenually make the dataset smaller while keeping the relevant data.

Reviewer: ```suggestion
We made the preprocessing more aggressive and implemented more preprocessing methods, to continually make the dataset smaller while keeping the relevant data.
```

> We also implemented thresholds to the $\theta$ and $beta$ matrixes made by the \gls{lda} model. All values under the normal destribution values were set to 0, so that we could work more efficiently with sparse matrixes.

Reviewer: ```suggestion
We also implemented thresholds to the $\theta$ and $beta$ matrixes made by the \gls{lda} model. 
All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrixes.
```

This ended up causing more trouble than it was worth, as we now had to deal with a large portion of our evaluations containing no values.
We also tried many different ways to construct the adjacency matrix more efficiently, described further in \autoref{app:adj_matrix}.

> However even after we managed to produce an adjecency matrix our results proved to be very bad.

Reviewer: ```suggestion
However even after we managed to produce an adjacency matrix our results proved to be very bad.
```

From here we began searching for more similar papers, where we managed to find two: \todo{refs}.
We focused particularly on \todo{ref}, due to a strong similarity between their project and our ideas.

> We used their methods to improve upon several parts of our methods, and we underwent change from building a single framework, to testing and comparing multiple different methods. However still our results were bad.

Reviewer: ```suggestion
We used their methods to improve upon several parts of our methods, and we changed from building a single framework, to testing and comparing multiple different methods. 
However, our results were still bad.
```

\subsection{Project Process}
\begin{itemize}
	\item Initial Idea - Combine LDA and PageRank (clustering)
	\item Preprocessing \& LDA implementation
	\item Adjacency matrix construction
	\begin{itemize}
		\item challenges with calculation time
		\item focus on more restricting preprocessing and thresholds
	\end{itemize}
	\item Bad results
	\item Discovery of related papers
	\item Found out our model didn't converge
\end{itemize}

Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the information retrieval results of \gls{lda}.

> We began by implementing some initial preprocessing methods and using an \gls{lda} implementation by gensim\todo{ref}.

Reviewer: ```suggestion
We began by implementing some initial preprocessing methods and using an \gls{lda} implementation by gensim\footnote{https://radimrehurek.com/gensim/models/ldamodel.html}.
```

After this, we only needed to combine \gls{lda} with a PageRank implementation. 
But to do so, we first needed to construct an adjacency matrix. 
This is where we started to run into trouble as constructing a large adjacency matrix based on a similarity measure took an extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.
We made the preprocessing more aggressive and implemented more preprocessing methods, to continually make the dataset smaller while keeping the relevant data.
We also implemented thresholds to the $\theta$ and $beta$ matrixes made by the \gls{lda} model. 
All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrixes.
This ended up causing more trouble than it was worth, as we now had to deal with a large portion of our evaluations containing no values.
We also tried many different ways to construct the adjacency matrix more efficiently, described further in \autoref{app:adj_matrix}.

However even after we managed to produce an adjacency matrix our results proved to be very bad.

> From here we began searching for more similar papers, where we managed to find two: \todo{refs}.

Reviewer: ```suggestion
From here we began searching for more similar papers, where we found: \todo{refs}.
```

We focused particularly on \todo{ref}, due to a strong similarity between their project and our ideas.
We used their methods to improve upon several parts of our methods, and we changed from building a single framework, to testing and comparing multiple different methods. 

> However, our results were still bad.

Reviewer: er dette vores færdige model eller før vi fik en god model

Author: nej, først den helt i bunden er den færdige

> It was by then that we began testing our lda model and found out that it was unable to converge using only a single pass through the corpus. We then began testing hyperparameters for the implementation of lda, before once again finding hyperparameters for our lda dataset.

Reviewer: ```suggestion
It was by then that we began testing our \gls{lda} model and found out that it was unable to converge using only a single pass through the corpus. 
We then began testing hyperparameters for the implementation of \gls{lda}, before once again finding hyperparameters for our \gls{lda}.
```

	\begin{itemize}
		\item challenges with calculation time
		\item focus on more restricting preprocessing and thresholds
	\end{itemize}
	\item Bad results
	\item Discovery of related papers
	\item Found out our model didn't converge
\end{itemize}

Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the information retrieval results of \gls{lda}.

We began by implementing some initial preprocessing methods and using an \gls{lda} implementation by gensim\footnote{https://radimrehurek.com/gensim/models/ldamodel.html}.

After this, we only needed to combine \gls{lda} with a PageRank implementation. 
But to do so, we first needed to construct an adjacency matrix. 
This is where we started to run into trouble as constructing a large adjacency matrix based on a similarity measure took an extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.
We made the preprocessing more aggressive and implemented more preprocessing methods, to continually make the dataset smaller while keeping the relevant data.

> We also implemented thresholds to the $\theta$ and $beta$ matrixes made by the \gls{lda} model. 

Author: ```suggestion
We also implemented thresholds to the $\theta$ and $beta$ matrices made by the \gls{lda} model.
This was also done to make sure that every document wasn't connected to every other document in the adjacency matrix.
```

All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrixes.

> This ended up causing more trouble than it was worth, as we now had to deal with a large portion of our evaluations containing no values.

Author: ```suggestion
This change ended up causing a lot more problems later on, as we now had to deal with a large portion of our evaluations containing no values. \todo{ref to related problems}
```

\subsection{Project Process}
\begin{itemize}
	\item Initial Idea - Combine LDA and PageRank (clustering)
	\item Preprocessing \& LDA implementation
	\item Adjacency matrix construction
	\begin{itemize}
		\item challenges with calculation time
		\item focus on more restricting preprocessing and thresholds
	\end{itemize}
	\item Bad results
	\item Discovery of related papers
	\item Found out our model didn't converge

> \end{itemize}

Author: ved ikke om vi skal have denne med?

Reviewer 2: Hmm, det er måske dobbeltkonfekt, men den giver alligevel et hurtigt overblik. Der burde nok i hvert fald være en linje efter itemizen der siger f.eks. "This process will now be described further.".

Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the information retrieval results of \gls{lda}.

We began by implementing some initial preprocessing methods and using an \gls{lda} implementation by gensim\footnote{https://radimrehurek.com/gensim/models/ldamodel.html}.

After this, we only needed to combine \gls{lda} with a PageRank implementation. 
But to do so, we first needed to construct an adjacency matrix. 
This is where we started to run into trouble as constructing a large adjacency matrix based on a similarity measure took an extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.
We made the preprocessing more aggressive and implemented more preprocessing methods, to continually make the dataset smaller while keeping the relevant data.
We also implemented thresholds to the $\theta$ and $beta$ matrices made by the \gls{lda} model.
This was also done to make sure that every document wasn't connected to every other document in the adjacency matrix.

> All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrixes.

Reviewer 2: ```suggestion
All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrices.
```

> All values under the normal distribution values were set to 0 so that we could work more efficiently with sparse matrixes.

Reviewer 2: Jeg kan ikke huske specifikt hvad vi gjorde, men normal distribution values lyder ikke helt rigtigt.

Author: det svarer også på til 1/længden af vectoren, det kan være det er bedre at skrive?
det er ihvertfald sådan jeg husker det..

Reviewer 2: Ja, det kunne være bedre hvis det er hvad vi gjorde. Så skal det bare være tydeligt hvilken vector det er.

\end{itemize}

Each of these items will now be described in more detail.

Our initial idea with the project was to combine \gls{lda} with a PageRank algorithm to see if this could improve the \acrlong{ir} results of \gls{lda}.
Originally we had wanted to use Clustering PageRank as described in \citeauthor{ClusterPageRank}\cite{ClusterPageRank}, but that idea was later scrapped.

We began by implementing some initial preprocessing methods and using an \gls{lda} implementation from Gensim\footnote{\url{https://radimrehurek.com/gensim/models/ldamodel.html}}.

After this, we only needed to combine \gls{lda} with a PageRank implementation. 
But to do so, we first needed to construct an adjacency matrix.
This is where we started to run into trouble as constructing a large adjacency matrix based on a similarity measure took an extremely long time, which forced us to make changes to the way we constructed the adjacency matrix as well as changes to the earlier phases: preprocessing and \gls{lda}.
We made the preprocessing more aggressive and implemented more preprocessing methods, to continually make the dataset smaller while keeping the relevant data.

> Importantly we also implemented thresholds to the topic-document distributions matrix $\theta$ and  the topic-word distributions matrix $\beta$ made by the \gls{lda} model.

Reviewer 2: ```suggestion
Importantly, we also implemented thresholds to the topic-document distribution matrix $\theta$ and the topic-word distribution matrix $\beta$ made by the \gls{lda} model.
```

For each distribution we replaced values below one divided by the number of values in the distribution, with 0.
This was also done to make sure that every document wasn't connected to every other document in the adjacency matrix, and to save calculation time by working with sparse matrices.
This change ended up causing a lot more problems later on, as many future calculations expected non-zero values, and would be dominated by zeros because of multiplication.

> Lastly we also tried many different ways to construct the adjacency matrix more efficiently, described further in \autoref{app:adj_matrix}.

Reviewer 2: ```suggestion
Lastly, we also tried many different ways to construct the adjacency matrix more efficiently, described further in \autoref{app:adj_matrix}.
```
